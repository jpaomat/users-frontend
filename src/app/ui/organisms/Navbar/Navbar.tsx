import PropTypes from 'prop-types';
import { INavbar } from '../../interfaces/organisms.model';
import { NavbarAuth, NavbarNav, navbarLinks } from '../../molecules';

export const Navbar = ({ type, size }: INavbar) => {
	return (
		<nav
			className={`navbar navbar-expand-sm navbar-${type} ${size}-${type} p-2`}
		>
			<div className='container-fluid'>
				<NavbarNav navbarItems={navbarLinks} />
			</div>
			<div className='collapse navbar-collapse w-100 dual-collapse2 d-flex justify-content-end'>
				<NavbarAuth userName='Paola' />
			</div>
		</nav>
	);
};

Navbar.defaultProps = {
	type: '',
	size: '',
};

Navbar.propTypes = {
	type: PropTypes.string,
	size: PropTypes.string,
};
