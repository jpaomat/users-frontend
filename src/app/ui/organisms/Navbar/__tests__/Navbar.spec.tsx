import { render } from '@testing-library/react';
import { Navbar } from '../Navbar';
import { MemoryRouter } from 'react-router-dom';

describe('<Navbar /> component', () => {

	beforeEach(()=> jest.clearAllMocks);
	
	test('should render component', () => {
		const { container } = render(
			<MemoryRouter>
				<Navbar />
			</MemoryRouter>
		);
		expect(container).toMatchSnapshot();
	});
});
