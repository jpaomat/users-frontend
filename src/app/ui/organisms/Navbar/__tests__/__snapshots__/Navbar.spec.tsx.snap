// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> component should render component 1`] = `
<div>
  <nav
    class="navbar navbar-expand-sm navbar- - p-2"
  >
    <div
      class="container-fluid"
    >
      <div
        class="collapse navbar-collapse"
        id="navbarNav"
      >
        <ul
          class="navbar-nav"
        >
          <a
            class="nav-link "
            href="/users"
          >
            Usuarios
          </a>
          <a
            class="nav-link "
            href="/posts"
          >
            Posts
          </a>
          <a
            class="nav-link "
            href="/albums"
          >
            √Ålbumes
          </a>
        </ul>
      </div>
    </div>
    <div
      class="collapse navbar-collapse w-100 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        class="navbar-nav ml-auto"
      >
        <span
          class="nav-item nav-link text-primary"
        >
          Paola
        </span>
        <button
          class="btn btn-outline-light btn-sm"
        >
          <span>
            Logout
          </span>
        </button>
      </ul>
    </div>
  </nav>
</div>
`;
